src/MLJDecisionTreeInterface.jl

```diff
@@ -260,6 +396,26 @@ function MMI.predict(::RandomForestRegressor, forest, Xnew)
     return DT.apply_forest(forest, Xmatrix)
 end
 
+MMI.reports_feature_importances(::Type{<:RandomForestRegressor}) = true
+
+function MMI.feature_importances(m::RandomForestRegressor, fitresult, report)
+    # generate feature importances for report
+    if m.feature_importance == :impurity
+        feature_importance_func = DecisionTree.impurity_importance
+    elseif m.feature_importance == :split
+        feature_importance_func = DecisionTree.split_importance
+    end
+
+    forest = fitresult
+    features = report.features
+    fi = feature_importance_func(forest)
+    fi_pairs = collect(Dict(zip(features, fi)))
+    # sort descending
+    sort!(fi_pairs, by= x->-x[2])
+
+    return fi_pairs
+end
+
```

Comment:
The code you have here is similar to that of `DecisionTreeClassifier`, `RandomForestRegressor` etc.. Why not combine them into a single function `MMI.function MMI.feature_importances(m::Union{DecisonTreeClassifier, RandomForestClassifier, RandomForestRegressor}, fitresult, report)`
to avoid code duplication.

Reply:
I left them separate as `fitresult` is not consistent for each method. For example, `DecisionTreeClassifier` has `    fitresult = (tree, classes_seen, integers_seen, features)` while `RandomForestRegressor` has `fitresult = forest`. I can add an if statement to check the length of the fitresult if we want to remove the duplicates. Would that work?

Reply:
Yes. Or you could define an fucntion like `get_fitresult(model::RandomForestRegressor, fitresult)` and ``get_fitresult(model::Union{DecisionTreeClassifier, ...}, fitresult)` that would extract the actual fitresult. You can then use this function in your code.